File: PHP/FindMostFrequentVowelAndConsonantTest.php
<?php
/**
 * @version PHP 8.2.22
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit FindMostFrequentVowelAndConsonantTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("FindMostFrequentVowelAndConsonant.php");

class FindMostFrequentVowelAndConsonantTest extends TestCase {

    public function testFindMostFrequentVowelAndConsonant() {
        $obj = new FindMostFrequentVowelAndConsonant();
        //Test 1
        $want = 6;
        $got = $obj->maxFreqSum( "successes" );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 3;
        $got = $obj->maxFreqSum( "aeiaeia" );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/FindMostFrequentVowelAndConsonant_test.go
/**
 * @version GO 1.23.5
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestFindMostFrequentVowelAndConsonant
 */
package GO

import (
	"reflect"
	"testing"
)

func TestFindMostFrequentVowelAndConsonant(t *testing.T) {
	tables := []struct {
		want    int
		letters string
	}{
		{6, "successes"},
		{3, "aeiaeia"},
	}

	for _, table := range tables {
		got := maxFreqSum(table.letters)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %b but the return was this: %b", table.want, got)
		}
	}
}
